eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -201,6 +237,35 @@ object Helpers {
     extractShutdownScript(accept.temporaryChannelId, localFeatures, remoteFeatures, accept.upfrontShutdownScript_opt).map(script_opt => (channelFeatures, script_opt))
   }
 
+  /** Called by the initiator of a dual-funded channel. */
+  def validateParamsInitiator(nodeParams: NodeParams, channelType: SupportedChannelType, localFeatures: Features[InitFeature], remoteFeatures: Features[InitFeature], open: OpenDualFundedChannel, accept: AcceptDualFundedChannel): Either[ChannelException, (ChannelFeatures, Option[ByteVector])] = {
+    validateChannelType(open.temporaryChannelId, channelType, open.channelType_opt, accept.channelType_opt, localFeatures, remoteFeatures) match {
+      case Some(t) => return Left(t)
+      case None => // we agree on channel type
+    }
+
+    if (accept.maxAcceptedHtlcs > Channel.MAX_ACCEPTED_HTLCS) return Left(InvalidMaxAcceptedHtlcs(accept.temporaryChannelId, accept.maxAcceptedHtlcs, Channel.MAX_ACCEPTED_HTLCS))
+
+    if (accept.dustLimit < Channel.MIN_DUST_LIMIT) return Left(DustLimitTooSmall(accept.temporaryChannelId, accept.dustLimit, Channel.MIN_DUST_LIMIT))
+    if (accept.dustLimit > nodeParams.channelConf.maxRemoteDustLimit) return Left(DustLimitTooLarge(open.temporaryChannelId, accept.dustLimit, nodeParams.channelConf.maxRemoteDustLimit))
+
+    // if minimum_depth is unreasonably large:
+    // MAY reject the channel.
+    if (accept.toSelfDelay > Channel.MAX_TO_SELF_DELAY || accept.toSelfDelay > nodeParams.channelConf.maxToLocalDelay) return Left(ToSelfDelayTooHigh(accept.temporaryChannelId, accept.toSelfDelay, nodeParams.channelConf.maxToLocalDelay))
+
+    val channelFeatures = ChannelFeatures(channelType, localFeatures, remoteFeatures)
+    extractShutdownScript(accept.temporaryChannelId, localFeatures, remoteFeatures, accept.upfrontShutdownScript_opt).map(script_opt => (channelFeatures, script_opt))
+  }
+
+  /** Compute the channelId of a dual-funded channel. */
+  def computeChannelId(open: OpenDualFundedChannel, accept: AcceptDualFundedChannel): ByteVector32 = {
+    if (LexicographicalOrdering.isLessThan(open.revocationBasepoint.value, accept.revocationBasepoint.value)) {
+      Crypto.sha256(open.revocationBasepoint.value ++ accept.revocationBasepoint.value)
+    } else {
+      Crypto.sha256(accept.revocationBasepoint.value ++ open.revocationBasepoint.value)
+    }
```

Comment:
Alternatively:
```suggestion
    val bin = Seq(open.revocationBasepoint.value, accept.revocationBasepoint.value)
      .sortWith(LexicographicalOrdering.isLessThan) // sort
      .reduce(_ ++ _) // concatenate
    Crypto.sha256(bin)
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/2247/commits/ab71773b7349191602c43e6b3e175fab8d06f159